### PR TITLE
add twitter search

### PR DESCRIPTION
**Pull request by jaxxstorm**
This attempts to add a twitter based search.

I'm not totally able to test this as I don't know how to provision the db and create my own bot, but would appreciate any feedback!

### CODE REVIEW

File: packages/apps/cron/src/searcher/twitter-searcher.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {ISearcher} from '.';
+import {Client} from 'twitter-api-sdk';
+
+class TwitterSearcher implements ISearcher {
+  t: Client;
+
+  constructor() {
+    this.t = new Client(process.env.BEARER_TOKEN);
```
Comment (manojVivek) [inline packages/apps/cron/src/searcher/twitter-searcher.ts:8]: Getting a bearer token, involves applying to Twitter for API access and approval.
Just a thought, is there a way to not getting into this? Like scrapping the information from web, as we're just focussed on the public search here.

Wdyt?

Comment (jaxxstorm) [reply packages/apps/cron/src/searcher/twitter-searcher.ts:8]: @manojVivek I think that's only for the stream API. If you're only using the public search endpoints, you can just generate a standard bearer token by signing up for a developer account:

image


https://developer.twitter.com/en/docs/authentication/oauth-2-0

I just did this, and there was no approval needed. It gives me 500k per month

image

Comment (manojVivek) [reply packages/apps/cron/src/searcher/twitter-searcher.ts:8]: Ok then, I'll sign up and test it in a few days and get back.

**Comment by jaxxstorm** [comment]
@manojVivek any updates?

**Comment by jaxxstorm** [comment]
ping @manojVivek

**Comment by manojVivek** [comment]
@jaxxstorm Apologies for the delay. 🙇🏼‍♂️ 

I just tested it, it looks good except for the undefined `author_id` in the URL. I have fixed it with #3.
Screenshot 2023-01-07 at 9 54 31 AM


So, merging this in now, thank you so much!  ❤️

**Comment by jaxxstorm** [comment]
Amazing! @manojVivek when will this roll out to all the deployments?

**Comment by jaxxstorm** [comment]
Looks like it's already working great! Thanks for merging!

**Pull request merged**